lib/metabox-api.php

```diff
@@ -0,0 +1,383 @@
+		 * @param WP_Post $post Post object.
+		 */
+		do_action( 'post_edit_form_tag', $post );
+
+		$referer = wp_get_referer();
+		?>><!-- End of Post Form Tag. -->
+		<?php wp_nonce_field( $nonce_action ); ?>
+		<?php
+			$current_user = wp_get_current_user();
+			$user_ID = $current_user->ID;
+		?>
+		<input type="hidden" id="user-id" name="user_ID" value="<?php echo (int) $user_ID ?>" />
+		<input type="hidden" id="hiddenaction" name="action" value="<?php echo esc_attr( $form_action ) ?>" />
+		<input type="hidden" id="originalaction" name="originalaction" value="<?php echo esc_attr( $form_action ) ?>" />
+		<input type="hidden" id="post_author" name="post_author" value="<?php echo esc_attr( $post->post_author ); ?>" />
+		<input type="hidden" id="post_type" name="post_type" value="<?php echo esc_attr( $post->post_type ) ?>" />
+		<input type="hidden" id="original_post_status" name="original_post_status" value="<?php echo esc_attr( $post->post_status ) ?>" />
+		<input type="hidden" id="referredby" name="referredby" value="<?php echo $referer ? esc_url( $referer ) : ''; ?>" />
+		<!-- This field is not part of the standard post form and is used to signify this is a gutenberg metabox. -->
+		<input type="hidden" name="gutenberg_metaboxes" value="gutenberg_metaboxes" />
+		<?php if ( ! empty( $active_post_lock ) ) { ?>
+		<input type="hidden" id="active_post_lock" value="<?php echo esc_attr( implode( ':', $active_post_lock ) ); ?>" />
+
+		<?php
+		}
+		if ( 'draft' != get_post_status( $post ) ) {
+			wp_original_referer_field( true, 'previous' );
+		}
+
+		echo $form_extra;
+
+		wp_nonce_field( 'meta-box-order', 'meta-box-order-nonce', false );
+		wp_nonce_field( 'closedpostboxes', 'closedpostboxesnonce', false );
+
+		// Permalink title nonce.
+		wp_nonce_field( 'samplepermalink', 'samplepermalinknonce', false );
+
+		/**
+		 * Fires at the beginning of the edit form.
+		 *
+		 * At this point, the required hidden fields and nonces have already been output.
+		 *
+		 * @since 3.7.0
+		 *
+		 * @param WP_Post $post Post object.
+		 */
+		do_action( 'edit_form_top', $post );
+
+		// Rendering the metaboxes.
+
+		// Styles.
+		$heading_style = "display:flex;justify-content:space-between;align-items:center;font-size:14px;margin:0;line-height: 50px;height: 50px;padding: 0 15px;background-color: #eee;border-top: 1px solid #e4e2e7;border-bottom: 1px solid #e4e2e7;box-sizing: border-box;";
+
+		/**
+		 * The #poststuff id selector is import for styles and scripts.
+		 *
+		 * It would be interesting to do a svn blame for that one. #blamenacin.
+		 * The sidebar-open class is used to work in tandem with the sidebar
+		 * opening and closing in the block editor. By default it is open.
+		 */
+		?>
+		<header class="gutenberg-metaboxes__header" style="<?php echo $heading_style; ?>">
+			<h2 style="font-size: 14px;">Extended Settings</h2>
+			<input name="save" type="submit" class="button button-primary button-large" id="publish" value="Update Settings">
```

Comment:
Can we move this out of the iframe, catch the click, submit the iframe and maybe refetch the post object to the Gutenberg state on success?

Reply:
I think so. Definitely worth a try.